From 51359b9df557853447192338283a6768184b6ffd Mon Sep 17 00:00:00 2001
From: Justin Cohen <justincohen@google.com>
Date: Tue, 5 Jun 2018 15:30:25 +0000
Subject: [PATCH] [ios] Various build fixes for Xcode 10.

  * Removes IDEBundleInjection.framework from egtests.
  * Corrects the DTXcode generation function to handle leading '10'.
  * Fixes a main_application_delegate SDK change
  * Fixes a non-null SDK change in a net unittest.

Bug: 843234, 849676
TBR: davidben@chromium.org
Cq-Include-Trybots: luci.chromium.try:ios-simulator-full-configs;master.tryserver.chromium.mac:ios-simulator-cronet
Change-Id: I6555bd396e4b4bb4d05822ee5867f878d7928c65
Reviewed-on: https://chromium-review.googlesource.com/1086233
Commit-Queue: Justin Cohen <justincohen@chromium.org>
Reviewed-by: Rohit Rao <rohitrao@chromium.org>
Reviewed-by: Sylvain Defresne <sdefresne@chromium.org>
Reviewed-by: Nico Weber <thakis@chromium.org>
Cr-Original-Commit-Position: refs/heads/master@{#564512}
Cr-Mirrored-From: https://chromium.googlesource.com/chromium/src
Cr-Mirrored-Commit: 27f9cbd88beea37bda9d6b46dabec9cf5eb386f3
---
 config/ios/rules.gni   | 13 +++++++++----
 config/mac/sdk_info.py | 34 +++++++++++++++++++++++++++-------
 2 files changed, 36 insertions(+), 11 deletions(-)

diff --git a/config/ios/rules.gni b/config/ios/rules.gni
index df6033bd7..7fdc93120 100644
--- a/config/ios/rules.gni
+++ b/config/ios/rules.gni
@@ -1805,10 +1805,15 @@ template("ios_xctest_test") {
     # Xcode needs those two framework installed in the application (and signed)
     # for the XCTest to run, so install them using extra_system_frameworks.
     _ios_platform_library = "$ios_sdk_platform_path/Developer/Library"
-    extra_system_frameworks = [
-      "$_ios_platform_library/Frameworks/XCTest.framework",
-      "$_ios_platform_library/PrivateFrameworks/IDEBundleInjection.framework",
-    ]
+    extra_system_frameworks =
+        [ "$_ios_platform_library/Frameworks/XCTest.framework" ]
+
+    # TODO: Remove this once support for Xcode 9.x is dropped.
+    if (xcode_version_int < 1000) {
+      extra_system_frameworks += [
+        "$_ios_platform_library/PrivateFrameworks/IDEBundleInjection.framework",
+      ]
+    }
 
     _xctest_bundle = _xctest_target + "_bundle"
     if (current_toolchain == default_toolchain) {
diff --git a/config/mac/sdk_info.py b/config/mac/sdk_info.py
index 8a9edc147..46dcec870 100644
--- a/config/mac/sdk_info.py
+++ b/config/mac/sdk_info.py
@@ -3,6 +3,8 @@
 # found in the LICENSE file.
 
 import argparse
+import doctest
+import itertools
 import os
 import subprocess
 import sys
@@ -10,16 +12,32 @@ import sys
 # This script prints information about the build system, the operating
 # system and the iOS or Mac SDK (depending on the platform "iphonesimulator",
 # "iphoneos" or "macosx" generally).
-#
-# In the GYP build, this is done inside GYP itself based on the SDKROOT
-# variable.
+
+def SplitVersion(version):
+  """Splits the Xcode version to 3 values.
+
+  >>> list(SplitVersion('8.2.1.1'))
+  ['8', '2', '1']
+  >>> list(SplitVersion('9.3'))
+  ['9', '3', '0']
+  >>> list(SplitVersion('10.0'))
+  ['10', '0', '0']
+  """
+  version = version.split('.')
+  return itertools.islice(itertools.chain(version, itertools.repeat('0')), 0, 3)
 
 def FormatVersion(version):
-  """Converts Xcode version to a format required for Info.plist."""
-  version = version.replace('.', '')
-  version = version + '0' * (3 - len(version))
-  return version.zfill(4)
+  """Converts Xcode version to a format required for DTXcode in Info.plist
 
+  >>> FormatVersion('8.2.1')
+  '0821'
+  >>> FormatVersion('9.3')
+  '0930'
+  >>> FormatVersion('10.0')
+  '1000'
+  """
+  major, minor, patch = SplitVersion(version)
+  return ('%2s%s%s' % (major, minor, patch)).replace(' ', '0')
 
 def FillXcodeVersion(settings):
   """Fills the Xcode version and build number into |settings|."""
@@ -53,6 +71,8 @@ def FillSDKPathAndVersion(settings, platform, xcode_version):
 
 
 if __name__ == '__main__':
+  doctest.testmod()
+
   parser = argparse.ArgumentParser()
   parser.add_argument("--developer_dir", required=False)
   args, unknownargs = parser.parse_known_args()
-- 
2.16.1

